host/sgx/create.c

```diff
@@ -727,6 +727,9 @@ oe_result_t oe_terminate_enclave(oe_enclave_t* enclave)
     if (!enclave || enclave->magic != ENCLAVE_MAGIC)
         OE_RAISE(OE_INVALID_PARAMETER);
 
+    /* stop the switchless manager */
+    oe_switchless_manager_shutdown(enclave);
+
```

Comment:
Can the enclave destructor make switchless ecalls? If so, the switchless-manager must be shutdown later.

Reply:
@EMumau 
Here is some information from anakrish
https://github.com/openenclave/openenclave/blob/1437683b73ac540d5467687e268f341485251d90/enclave/core/sgx/calls.c#L444-L460

An enclave has an init and a corresponding destructor
At the time of destruction it can make switchless ecalls. 
It is safest to shutdown the switchless manager after the destruction of the enclave.